<?xml version="1.0"?>
<launch>
    <!--Launch Sensor Node-->
	<!--include file="$(find sq1_extra)/launch/run_sq2_for_joy.launch"/-->

    <!--Remove Ground-->
    <!--node pkg="sensor_fusion" type="sq_rm_ground" name="sq_rm_ground">
		<remap from="/cloud" 	 to="/lcl/center" />
		<remap from="/rm_ground" to="/rm_ground/center" />
    </node-->

	<!--Plane Segmentation-->
    <param name="min_x" type="double" value="0"   />
    <param name="max_x" type="double" value="4.0" />
    <param name="min_y" type="double" value="-1.0"/>
    <param name="max_y" type="double" value="1.0" />
    <param name="min_z" type="double" value="-0.6"/>
    <param name="max_z" type="double" value="1.0" />
	<node pkg="sensor_fusion" type="lidar_seg_plane" name="lidar_seg_plane" output="screen">
		<remap from="/cloud"        to="/rm_ground/center" />
		<remap from="/area"         to="/center/area" />
        <remap from="/down_sample"  to="/center/down_sample" />
        <remap from="/cluster"      to="/center/cluster" />
        <remap from="/plane"        to="/center/plane" />
        <remap from="/outlier"      to="/center/outlier" />
	</node>

    <!--Circle Detection-->
    <node pkg="sensor_fusion" type="lidar_seg_circle" name="lidar_seg_circle" output="screen">
        <remap from="/cloud"            to="/center/plane" />
        <remap from="/output/reference" to="/center/reference" />
        <remap from="/output/search"    to="/center/search" />]
        <remap from="/output/candidate" to="/center/candidate" />
        <remap from="/output/centroid"  to="/center/centroid" />
        <remap from="/output/average"   to="/center/average" />
    </node>

	<!--Dumy PointCloud-->
	<!-- node pkg="sensor_fusion" type="dumy_lidar" name="duny_lidar">
		<remap from="/cloud" to="/center/plane" />
		<remap from="/output" to="/dumy" />
	</node -->

	<!--PointCloud2Image-->
	<!--node pkg="sensor_fusion" type="pointcloud2image" name="pointcloud2image" >
		<remap from="/cloud" to="/center/plane" />
	</node-->
	
	<!--divide PointCloud-->
	<!--node pkg="sensor_fusion" type="detection_circle" name="detection_circle" >
		<remap from="/image" 	to="/zed0/left/image_rect_color/republish" />
	</node-->
	
</launch>
